tests/functional/nccl_message_transfer.c

```diff
@@ -75,7 +75,8 @@ int main(int argc, char* argv[])
 	/* Set CUDA device for subsequent device memory allocation, in case GDR is used */
 	cuda_dev = local_rank;
 	NCCL_OFI_TRACE(NCCL_NET, "Using CUDA device %d for memory allocation", cuda_dev);
-	CUDACHECK(cudaSetDevice(cuda_dev));
+	CUcontext primary_context;
+	CUDACHECK(cuDevicePrimaryCtxRetain(&primary_context, cuda_dev));
```

Comment:
I looked through the NCCL code to see where they create and set CUDA contexts, and found the function calls [here](https://github.com/NVIDIA/nccl/blob/b6d7438d3145a619f924dbbca6c96db21fab716e/src/proxy.cc#L810-L827). I decided to replace my call to `cuDevicePrimaryCtxRetain` with the same `cuCtxCreate` and `cuCtxSetCurrent` calls that NCCL makes so that if context creation fails in the functional test, it indicates that context creation in NCCL would fail as well.